### ISSUE TITLE
@MicronautTest doesn't have support for starting AWS lambda functions that extend the 'MicronautRequestHandler' class

### ISSUE THREAD

**Issue by athkalia** [micronaut-projects/micronaut-core#4556]
It seems like there is no easy way to instantiate a lambda function by using the `@MicronautTest` annotation.

In the AWS lambda function tutorials, e.g. here https://guides.micronaut.io/mn-serverless-function-aws-lambda-kotlin/guide/index.html it suggested that to test a handler you just instantiate it. That however, makes it difficult to start a handler with a specific environment, or other configuration that is typically passed via the '@MicronautTest' annotation.

If a `@MicronautTest' is defined and the handler is injected instead of created manually, we still get Micronaut starting up twice, as when the constructor of the `MicronautRequestHanlder` looks like this:

```
public MicronautRequestHandler() {
        buildApplicationContext(null);
        applicationContext.inject(this);
    }
```

The only way I could make this work was to pass a system property before running the test, like:

```
System.setProperty("micronaut.environments", "myenv")
val handler = MyHandlerRepoCheckoutHandler()
```

### Task List

- [ ] Steps to reproduce provided
- [ ] Stacktrace (if present) provided
- [ ] Example that reproduces the problem uploaded to Github
- [ ] Full description of the issue provided (see below)

### Steps to Reproduce

1. TODO
2. TODO
3. TODO

### Expected Behaviour

Tell us what should happen

### Actual Behaviour

Tell us what happens instead

**Comment by graemerocher** [comment]
yeah `@MicronautTest` is kind of incompatible with that approach right now, would be nice if there is a solution

**Comment by athkalia** [comment]
cc @ttzn
